presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -74,17 +74,19 @@ PrestoExchangeSource::PrestoExchangeSource(
     std::shared_ptr<exec::ExchangeQueue> queue,
     memory::MemoryPool* pool,
     const std::string& clientCertAndKeyPath,
-    const std::string& ciphers)
+    const std::string& ciphers,
+    std::shared_ptr<folly::IOExecutor> executor)
     : ExchangeSource(extractTaskId(baseUri.path()), destination, queue, pool),
       basePath_(baseUri.path()),
       host_(baseUri.host()),
       port_(baseUri.port()),
       clientCertAndKeyPath_(clientCertAndKeyPath),
       ciphers_(ciphers) {
   folly::SocketAddress address(folly::IPAddress(host_).str(), port_, true);
-  auto* eventBase = folly::getUnsafeMutableGlobalEventBase();
   auto timeoutMs = std::chrono::duration_cast<std::chrono::milliseconds>(
       SystemConfig::instance()->exchangeRequestTimeout());
+  executor_ = executor ? executor : folly::getUnsafeMutableGlobalIOExecutor();
```

Comment:
In which case this would be null? thanks!

Reply:
This is for tests, let me put this logic back to tests